smtp/client.js

```diff
@@ -1,209 +1,186 @@
-			   };
-		   })(this));
-	   } else {
-		   return this._isAttachmentInlinedHtml(attachment);
-	   }   
-	},
-
-   _isAttachmentInlinedHtml: function(attachment) {
-	   return attachment && 
-		  (attachment.data || attachment.path) && 
-		   attachment.alternative === true;
-   },
-
-   _sendsmtp: function(stack, next)
-   {
-      var self   = this;
-      var check= function(err)
-      {
-         if(!err && next)
-         {
-            next.apply(self, [stack]);
-         }
-         else
-         {
-            // if we snag on SMTP commands, call done, passing the error
-            // but first reset SMTP state so queue can continue polling
-            self.smtp.rset(function() { self._senddone(err, stack); });
-         }
-      };
-
-      return check;
-   },
-
-   _sendmail: function(stack)
-   {
-      var self = this;
-      var from = stack.returnPath || stack.from;
-      self.sending = true;
-      self.smtp.mail(self._sendsmtp(stack, self._sendrcpt), '<' + from + '>');
-   },
-
-   _sendrcpt: function(stack)
-   {
-      var self = this, to = stack.to.shift().address;
-      self.smtp.rcpt(self._sendsmtp(stack, stack.to.length ? self._sendrcpt : self._senddata), '<'+ to +'>');
-   },
-
-   _senddata: function(stack)
-   {
-      var self = this;
-      self.smtp.data(self._sendsmtp(stack, self._sendmessage));
-   },
-
-   _sendmessage: function(stack)
-   {
-      var self = this, stream = stack.message.stream();
-
-      stream.on('data', function(data) { self.smtp.message(data); });
-      stream.on('end', function() { self.smtp.data_end(self._sendsmtp(stack, function() { self._senddone(null, stack) })); });
-
-      // there is no way to cancel a message while in the DATA portion, so we have to close the socket to prevent
-      // a bad email from going out
-      stream.on('error', function(err) { self.smtp.close(); self._senddone(err, stack); });
-   },
-
-   _senddone: function(err, stack)
-   {
-      var self = this;
-      self.sending = false;
-      stack.callback(err, stack.message);
-      self._poll();
-   }
-};
+const smtp = require('./smtp');
+const message = require('./message');
+const addressparser = require('addressparser');
+
+class Client {
+	constructor(server) {
+		this.smtp = new smtp.SMTP(server);
+		//this.smtp.debug(1);
+
+		this.queue = [];
+		this.timer = null;
+		this.sending = false;
+		this.ready = false;
+	}
+
+	_poll() {
+		clearTimeout(this.timer);
+
+		if (this.queue.length) {
+			if (this.smtp.state() == smtp.state.NOTCONNECTED) {
+				this._connect(this.queue[0]);
+			} else if (
+				this.smtp.state() == smtp.state.CONNECTED &&
+				!this.sending &&
+				this.ready
+			) {
+				this._sendmail(this.queue.shift());
+			}
+		}
+		// wait around 1 seconds in case something does come in, otherwise close out SMTP connection if still open
+		else if (this.smtp.state() == smtp.state.CONNECTED)
+			this.timer = setTimeout(() => this.smtp.quit(), 1000);
```

Comment:
the eslint/prettier settings i use now force brackets around conditional clauses, even if only one line.

other areas in this code do include the brackets, so i'ld like to shoot for at least consistency. 

it seems like you are integrating eslint/prettier with this PR. did prettier run on this file? or is there a rule that would be needed to turn on to get the brackets to be injected?

i think i am used to the airbnb recommendations, so not sure. what do you think?